crates/ra_ide/src/snapshots/highlight_unsafe.html

```diff
@@ -37,12 +37,23 @@
     <span class="keyword unsafe">unsafe</span> <span class="keyword">fn</span> <span class="function declaration unsafe">unsafe_method</span>(&<span class="self_keyword">self</span>) {}
 }
 
+<span class="attribute">#[</span><span class="function attribute">repr</span><span class="attribute">(packed)]</span>
+<span class="keyword">struct</span> <span class="struct declaration">Packed</span> {
+    <span class="field declaration">a</span>: <span class="builtin_type">u16</span>,
+    <span class="field declaration">b</span>: <span class="builtin_type">u8</span>,
+    <span class="field declaration">c</span>: <span class="builtin_type">u32</span>,
+}
+
 <span class="keyword">fn</span> <span class="function declaration">main</span>() {
     <span class="keyword">let</span> <span class="variable declaration">x</span> = &<span class="numeric_literal">5</span> <span class="keyword">as</span> *<span class="keyword">const</span> <span class="builtin_type">usize</span>;
     <span class="keyword unsafe">unsafe</span> {
         <span class="function unsafe">unsafe_fn</span>();
         <span class="struct">HasUnsafeFn</span>.<span class="function unsafe">unsafe_method</span>();
         <span class="keyword">let</span> <span class="variable declaration">y</span> = <span class="operator unsafe">*</span><span class="variable">x</span>;
         <span class="keyword">let</span> <span class="variable declaration">z</span> = -<span class="variable">x</span>;
+        <span class="keyword">let</span> <span class="variable declaration">packed</span> = <span class="struct">Packed</span> { <span class="field">a</span>: <span class="numeric_literal">0</span>, <span class="field">b</span>: <span class="numeric_literal">0</span>, <span class="field">c</span>: <span class="numeric_literal">0</span> };
+        <span class="keyword">let</span> <span class="variable declaration">a</span> = <span class="operator unsafe">&</span><span class="variable">packed</span><span class="operator unsafe">.</span><span class="field">a</span>;
+        <span class="keyword">let</span> <span class="variable declaration">b</span> = <span class="operator unsafe">&</span><span class="variable">packed</span><span class="operator unsafe">.</span><span class="field">b</span>;
+        <span class="keyword">let</span> <span class="variable declaration">c</span> = <span class="operator unsafe">&</span><span class="variable">packed</span><span class="operator unsafe">.</span><span class="field">c</span>;
```

Comment:
I'm not sure why the `.` operator is also marked as unsafe. It makes sense since it's in combination with the ref, but this was unexpected.

Reply:
So I'm able to fix this by changing the expr match to be against `T![&]` rather than `REF_EXPR`. However, I realized we might also want to account for autoref'ed function calls. I added a test for that, and regardless of with expr match I use, it doesn't get highlighted. I added a test for it to show we don't currently include it. Should I try to work on getting support for that in this PR?

Reply:
Also, should this include `ref` expressions within patterns?

Reply:
>I'm not sure why the . operator is also marked as unsafe. 

I think I've fixed this in master some time ago

Reply:
Done